### ISSUE TITLE
MRBS 1.11.4: Limit r/w access to rooms

### ISSUE THREAD

**Issue by TheDude1969** [meeting-room-booking-system/mrbs-code#3644]
Hello Community,

is there maybe a way to allow bookings to defined room by membership of an AD security group?

I am using LDAP for authentication and want to use security groups to control access to rooms but was unable to find something about this in the documentation.

Many thanks in advance,
best regards

TheDude

**Comment by campbell-m** [comment]
You could try using [the roles branch](https://github.com/meeting-room-booking-system/mrbs-code/tree/roles), which allows roles to be assigned to users and groups.  For each role you can define which rooms and areas are visible, bookable or admin-able.

The code is still under development though and it will also upgrade your database, after which it won't work with 1.11.4.   So I recommend you try it out first on a copy of your database.

**Comment by KawaMood** [comment]
> You could try using [the roles branch](https://github.com/meeting-room-booking-system/mrbs-code/tree/roles), which allows roles to be assigned to users and groups. For each role you can define which rooms and areas are visible, bookable or admin-able.
> 
> The code is still under development though and it will also upgrade your database, after which it won't work with 1.11.4. So I recommend you try it out first on a copy of your database.

Hi there, I have a question concerning this. 
If I start to use this role branch now, would I be able to upgrade MRBS when this feature will be part of an official release?

**Comment by campbell-m** [comment]
Yes, when this feature eventually becomes part of an official release you'll be able to upgrade your installation.  In the meantime new features and bug fixes applied to the main branch are also merged with the roles branch.

**Comment by coxmunro** [comment]
> > You could try using [dagkonferens i stockholm](https://konferensbokarna.se/stockholm), which allows roles to be assigned to users and groups. For each role you can define which rooms and areas are visible, bookable or admin-able.
> > The code is still under development though and it will also upgrade your database, after which it won't work with 1.11.4. So I recommend you try it out first on a copy of your database.
> 
> Hi there, I have a question concerning this. If I start to use this role branch now, would I be able to upgrade MRBS when this feature will be part of an official release?

I had the same question in my mind, now got the answer. Thanks

**Comment by zaid4840** [comment]
If you start using the role branch now, you will be able to upgrade MRBS when this feature becomes part of an official release. However, it is recommended to test it on a copy of your database first, as the code is still under development and may upgrade your database in a way that is not compatible with version 1.11.4.

For more information and to explore our SEO services, please visit our website: [HZA SEO Services][]((https://www.hzaseoservices.com/).)

**Comment by wohenbushuang** [comment]
Hi there, it has been four years after the roles branch [established](https://github.com/meeting-room-booking-system/mrbs-code/commit/732850f23b3ebf50b3630c79b26c23eccfeddbf9), would you consider to provide an official basical release version first? Maybe done is better than perfect.

**Comment by campbell-m** [comment]
Well there's nothing to stop you using it - I'm not quite sure what benefit a release brings.

There's still some more work to be done to finish it off.  However what I have been considering as an idea is to:

- make a 1.11.6 release of MRBS in the next month or two
- add a configuration setting, eg `$enable_roles`, which by default is false and make the roles branch behave in exactly the same way as standard MRBS.
- then merge ther roles branch back into main
- then make another release; if you want to use roles just set the config setting to true.

**Comment by wohenbushuang** [comment]
> The code is still under development though and it will also upgrade your database

That's what I'm afraid. Actually I modified this awesome project to adapt with our existing oauth2 authorization system. I'm happy to migrate the oauth2 part based on this role feature in someday when it is kind of "stable" (e.g. a release) meaning this role feature won't be abandon (because of too much dev time cost).

So glad to hear the future milestones and eagerly look forward to them.

**Comment by the-mark-hash** [comment]
For managing room bookings by AD security groups, using a role-based system like the one suggested is an excellent idea. At[ [Thoresta Herrgård](https://thoresta.se/offert-kontakt/hitta-hit/
![images (3) (1)](https://github.com/user-attachments/assets/72c47049-ea57-4b70-8392-1fcbdee0bb34)
), we focus on providing customized solutions for workshops and conferences, ensuring smooth booking experiences. Testing the feature on a backup database is highly recommended before full deployment to avoid disruptions. If you'd like to explore venues or tools that align with your needs, feel free to connect!

**Comment by covers-in-play** [comment]
Japanese swords are not only historical weapons but also valuable cultural artifacts that require proper knowledge and responsible ownership. Access to functional blades should be limited to trained practitioners, collectors, and enthusiasts who understand safety and authenticity. Brands like [samurai sword store](https://katokatana.com/) help educate buyers by providing reliable information about craftsmanship, usage, and care, ensuring that modern users respect the tradition and purpose behind these iconic swords.

**Comment by kbulygin** [comment]
> Trusted brands such as (link) also emphasize proper storage, safety, and education to preserve the quality and cultural value of these swords.

That's a creative ad. (See also the previous edit of the message.)

**Comment by campbell-m** [comment]
> That's a creative ad

Now deleted.
